### PR TITLE
Adding functionality for importing streams from history.

### PR DESCRIPTION
**Pull request by botronic**
## Description
Adds support for importing streams from history. As of right now streams cannot be cherry picked, either import all streams from the previous session or none. We could look at this later, it is kinda sugar sugar.

## Important files to review
main.py - All changes lie here
main.ui - Some additions to the add stream menu

## See also
Issue this addresses: https://github.com/kaszim/desktop-stream-viewer/issues/50

### CODE REVIEW

File: history.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+twitch.tv/esl_csgo
```
Comment (mictab) [inline history.txt]: Don't commit your own history :)

File: main.py [hunk 2]
```diff
@@ -57,6 +58,8 @@ def setup_ui(self):
         self.__bind_view_to_action(IMPORT_STREAMS_FROM_CLIPBOARD, self.import_streams_from_clipboard)
 
         self.recent_menu = self.ui.findChild(QtCore.QObject, "menuRecent")
+        self.ui.findChild(QtCore.QObject, LOAD_STREAM_HISTORY) \
```
Comment (mictab) [inline main.py]: You can use the `__bind_view_to_action` method here.

File: main.py [hunk 3]
```diff
@@ -264,6 +271,29 @@ def _get_user_quality_preference(self, stream_options):
...
         )
 
+    def stream_history(self):
+        """Starts streaming all streams that were playing when last session was closed."""
+        stream_history = self.load_stream_history()
+        if stream_history:
+            for stream in stream_history:
+                self.add_new_stream(stream)
+
+    def save_stream_to_history(self, url):
```
Comment (mictab) [inline main.py]: Can we do this in a separate thread?

File: history.txt (hunk 1 above)
Comment (botronic) [inline history.txt]: ops

File: main.py (hunk 3 above)
Comment (mictab) [inline main.py]: Since we're writing to a file and this could get messy, let's leave this for now.

File: main.py [hunk 4]
```diff
@@ -264,6 +270,29 @@ def _get_user_quality_preference(self, stream_options):
...
+        with open(HISTORY_FILE, 'a') as f:
+            f.write(url + '\n')
+
+    def load_stream_history(self):
+        """Loads up all streams from last session."""
+        stream_history = set()
+        with open(HISTORY_FILE, 'r') as f:
+            lines = f.readlines()
+            for line in lines:
+                stream_history.add(line.strip("\n"))
```
Comment (mictab) [inline main.py]: Let's just add a small comment here about what this line accomplishes.

Comment (mictab) [inline main.py]: The .close() line

**Pull request merged**